Packages/com.bci4kids.bciessentials/Runtime/Scripts/Utilities/ArrayUtilities.cs

```diff
@@ -81,10 +83,88 @@ void PopulateOptions()
             
             return randomizedOptions;
         }
+        #endregion
 
+        #region RNRA for Low Number of Targets and flashing
+        /// <summary>
+        /// This is an array to generate randomly flashing targets during the stimulus run for low number of targets.
+        /// This is NOT to be used for training targets. It will take the whole list and shuffle
+        /// only guaranteeing that the values will not repeat. It won't guarantee that all values
+        /// are presented first before repeating.
+        /// </summary>
+        /// 
+        /// <summary>
+        /// Generates an array with the given length populated using a MinHeap
+        /// to ensure weighted random selection and optimization.
+        /// </summary>
+        /// <param name="arrayLength">Number of values to include in the range.</param>
+        /// <param name="minRangeValue">The lowest value possible to include.</param>
+        /// <param name="maxRangeValue">The largest value possible to include.</param>
+        /// <returns>An int array</returns>
+        public static int[] GenerateWeightedArray(int arrayLength, int minRangeValue, int maxRangeValue)
```

Comment:
I am skeptical of this, what is its purpose? I don't see it being used anywhere.

Reply:
This is to be finished a bit later. Specifically, it's for low number SPOs, where we can use a Weighted Array flip instead of an RNRA to help create a more "random" feeling (i.e. for SPOs <3). Still working on it, forgot it was in there.

Reply:
Cool cool, we can cross that bridge later.

Reply:
Agreed, removed it for the time being.